GoogleColabTPU.ipynb

```diff
@@ -43,67 +56,66 @@
-     "  !wget https://github.com/TPU-Mersenne-Prime-Search/TensorPrime/archive/master.zip -O master.zip\n",
+     "  '''Download/Install/Configure TensorPrime.'''\n",
+     "  print('Downloading, building and setting up TensorPrime\\n')\n",
+     "  !wget -O master.zip https://github.com/TPU-Mersenne-Prime-Search/TensorPrime/archive/master.zip\n",
      "  !unzip -o master.zip\n",
      "  !mv TensorPrime-master/ tensorprime\n",
-     "  !wget https://raw.github.com/tdulcet/Distributed-Computing-Scripts/master/primenet.py -nv -O tensorprime/primenet.py\n",
+     "  !wget -nv -O tensorprime/primenet.py https://raw.github.com/tdulcet/Distributed-Computing-Scripts/master/primenet.py\n",
      "  # print(\"Registering computer with PrimeNet\\n\")\n",
      "  # TODO: This --tpu option would need to be added unless you integrate PrimeNet support into TensorPrime\n",
-     "  # !cd tensorprime && python3 primenet.py -d -t 0 -W 1 -T $type_of_work -u $prime_ID -i \"{'worktodo' + computer_number + '.txt'}\" -r \"{'results' + computer_number + '.txt'}\" -l \"{'local' + computer_number + '.ini'}\" --tpu -H $computer_name\n",
+     "  # !cd tensorprime && python3 primenet.py -OO -t 0 -W 1 -T {type_of_work} -u '{prime_ID}' -i '{'worktodo' + computer_number + '.txt'}' -r '{'results' + computer_number + '.txt'}' -l '{'local' + computer_number + '.ini'}' --tpu -H '{computer_name}'\n",
      "  run()\n",
      "\n",
      "def debug_exit():\n",
-     "  '''Output TPU and output of Prime95 or TensorPrime output'''\n",
+     "  '''Output TPU and output of Prime95 or TensorPrime output.'''\n",
      "  if debug == 'TPU (TensorPrime)' and os.path.exists(f'tensorprime/tpu{computer_number}.out'):\n",
-     "    print(f\"\\nOutput for computer number {computer_number}:\\n\")\n",
-     "    print(\"\\nPrimeNet output:\\n\")\n",
-     "    # !tail -n 100 \"{'tensorprime/primenet' + computer_number + '.out'}\" # view primenet output\n",
-     "    print(\"\\nTPU (TensorPrime) output: \")\n",
-     "    !tail -n 100 \"{'tensorprime/tpu' + computer_number + '.out'}\" # view TensorPrime progress\n",
-     "    # !cd tensorprime && python3 primenet.py -l \"{'local' + computer_number + '.ini'}\" -s\n",
+     "    print(f'\\nOutput for computer number {computer_number}:\\n')\n",
+     "    print('\\nPrimeNet output:\\n')\n",
+     "    # !tail -n 100 '{'tensorprime/primenet' + computer_number + '.out'}' # view primenet output\n",
+     "    print('\\nTPU (TensorPrime) output: ')\n",
+     "    !tail -n 100 '{'tensorprime/tpu' + computer_number + '.out'}' # view TensorPrime progress\n",
+     "    # !cd tensorprime && python3 -OO primenet.py -l '{'local' + computer_number + '.ini'}' -s\n",
      "    print()\n",
      "  else:\n",
-     "    print(f'No `{debug}` output file found for debug option and computer number `{computer_number}`.\\n')\n",
+     "    print(f'No `{debug!r}` output file found for debug option and computer number `{computer_number!r}`.\\n')\n",
```

Comment:
I've never seen the `!r` syntax, but neat.

Reply:
Yeah, I use the syntax fairly heavily in our PrimeNet script. Specifically, this is a [conversion flag](https://docs.python.org/3/library/string.html#formatstrings), which automatically calls `repr()` on the argument.